test/e2e/scheduling/preemption.go

```diff
@@ -430,19 +430,11 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
 			ginkgo.By("Verify there are 3 Pods left in this namespace")
 			wantPods := sets.NewString("high", "medium", "low")
 
-			var pods []v1.Pod
 			// Wait until the number of pods stabilizes. Note that `medium` pod can get scheduled once the
 			// second low priority pod is marked as terminating.
-			// TODO: exact the wait.PollImmediate block to framework.WaitForNumberOfRunningPods.
-			err := wait.PollImmediate(framework.Poll, framework.PollShortTimeout, func() (bool, error) {
-				podList, err := cs.CoreV1().Pods(ns).List(context.TODO(), metav1.ListOptions{})
-				// ignore intermittent network error
-				if err != nil {
-					return false, nil
-				}
-				pods = podList.Items
-				return len(pods) == 3, nil
-			})
+			framework.ExpectNoError(e2epod.WaitForPodsRunningReady(cs, ns, 3, 0, framework.PollShortTimeout, map[string]string{}))
```

Comment:
This change doesn't look equivalent/efficient comparing to original logic?

1. `e2epod.WaitForPodsRunningReady` semantically does "verifying pods are ready" (and even more); here we just verified whether the pods exist (even without verifying if they're ready)
2. `e2epod.WaitForPodsRunningReady` doesn't return pods, so you have to run `e2epod.GetPods` again to obtain the pods - which is inefficient.

Reply:
That is a good point.
Then do we just need to remove the TODO from the comment to avoid confusion in the future?

Reply:
SGTM. I'd suggest extracting the logic into a function called "WaitForNumberOfPods", and then remove the TODO comment.

Reply:
Thanks for review! Yes, this comment confused me. Fixed, as you suggested.